packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -279,6 +280,20 @@ export class Publisher {
             new Error(`[${response.status}] ${response.statusText}`)
           )
           return
+        } else if (response.status === 429) {
```

Comment:
This is a nice to have but I think it would be good to update the `http_request` event emitter to also give the current retryCount (and maybe the maxAttempts as well)!